js/researches/findTransitionWords.js

```diff
@@ -1,38 +1,55 @@
-var transitionWords = require( "../config/transitionWords.js" )();
-var twoPartTransitionWords = require( "../config/twoPartTransitionWords.js" )();
+var transitionWords = require( "../config/transitionWords.js" );
+var twoPartTransitionWords = require( "../config/twoPartTransitionWords.js" );
 var createRegexFromDoubleArray = require( "../stringProcessing/createRegexFromDoubleArray.js" );
 var getSentences = require( "../stringProcessing/getSentences.js" );
 var createRegexFromArray = require( "../stringProcessing/createRegexFromArray.js" );
-var filter = require( "lodash/filter" );
+var forEach = require( "lodash/forEach" );
 
 /**
  * Checks how many sentences from a text contain at least one transition word or two-part transition word
  * that are defined in the transition words config and two part transition words config.
- * @param {string} paper The Paper object to get text from.
+ * @param {Paper} paper The Paper object to get text from.
  * @returns {object} An object with the total number of sentences in the text
  * and the total number of sentences containing one or more transition words.
  */
 module.exports = function( paper ) {
-	var text = paper.getText();
-	var sentences = getSentences( text );
+	var sentences = getSentences( paper.getText() );
+
+	var sentenceResults = [];
 	var transitionWordSentenceCount = 0;
-	var twoPartTransitionWordRegex = createRegexFromDoubleArray( twoPartTransitionWords );
-	var transitionWordRegex = createRegexFromArray( transitionWords );
-	var nonMatchedSentences = filter( sentences, function ( sentence ) {
-		if ( sentence.match( twoPartTransitionWordRegex ) !== null ) {
+
+	var twoPartTransitionWordRegex = createRegexFromDoubleArray( twoPartTransitionWords() );
+	var transitionWordRegex = createRegexFromArray( transitionWords() );
+
+	forEach( sentences, function( sentence ) {
+		var twoPartMatches = sentence.match( twoPartTransitionWordRegex );
+
+		if ( twoPartMatches !== null ) {
+			sentenceResults.push( {
+				sentence: sentence,
+				transitionWords: twoPartMatches
+			} );
+
 			transitionWordSentenceCount++;
-			return false;
+			return;
 		}
-		return true;
-	} );
-	nonMatchedSentences.map( function( sentence ) {
-		if ( sentence.match( transitionWordRegex ) !== null ) {
+
+		var transitionWordMatches = sentence.match( transitionWordRegex );
+
+		if ( transitionWordMatches !== null ) {
+			sentenceResults.push( {
+				sentence: sentence,
+				transitionWords: transitionWordMatches
+			} );
+
 			transitionWordSentenceCount++;
+			return;
 		}
 	} );
 
 	return {
```

Comment:
We could put this return in the module.exports, and put the rest of the functionality in a separate function, since it is quite long.